src/bios.rs

```diff
@@ -142,10 +145,69 @@ impl Component for Bios {
         crate::component::query_adopt_state()
     }
 
+    // Backup the current grub.cfg and install new static config
+    // - Backup "/boot/loader/grub.cfg" to "/boot/grub2/grub.cfg.bak"
+    // - Remove symlink "/boot/grub2/grub.cfg"
+    // - Install new static grub.cfg
+    // - Remove old "/boot/loader/grub.cfg"
+    fn migrate_static_grub_config(&self, sysroot_path: &str, destdir: &openat::Dir) -> Result<()> {
+        let grub = "boot/grub2";
+        // sysroot_path is /, destdir is Dir of /
+        let grub_config_path = Utf8PathBuf::from(sysroot_path).join(grub);
+        let grub_config_dir = destdir.sub_dir(grub).context("Opening boot/grub2")?;
+
+        let grub_config = grub_config_path.join(grubconfigs::GRUBCONFIG);
+
+        // On BIOS, /boot/grub2/grub.cfg is symlink to /boot/loader/grub.cfg,
+        // backup it to /boot/grub2/grub.cfg.bak
+        if !grub_config.exists() {
+            anyhow::bail!("Could not find '{}'", grub_config);
+        }
+
+        let mut current_config;
+        if !grub_config.is_symlink() {
+            println!("'{}' is not a symlink", grub_config);
+            current_config = grub_config.clone();
+        } else {
+            // Get real file if /boot/grub2/grub.cfg is symlink
+            let real_config = grub_config_dir.read_link(grubconfigs::GRUBCONFIG)?;
+            let real_config =
+                Utf8PathBuf::from_path_buf(real_config).expect("Path should be valid UTF-8");
+            // Resolve symlink location
+            current_config = grub_config_path.clone();
+            current_config.push(real_config);
+        }
+
+        let backup_config = grub_config_path.join(grubconfigs::GRUBCONFIG_BACKUP);
+        if !backup_config.exists() {
+            // Backup the current GRUB config which is hopefully working right now
+            println!(
+                "Creating a backup of the current GRUB config '{}' in '{}'...",
+                current_config, backup_config
+            );
+            std::fs::copy(&current_config, &backup_config)
+                .context("Failed to backup GRUB config")?;
+        }
+
+        crate::grubconfigs::install(&destdir, None, true)?;
+
+        // Remove the real config if it is symlink
+        if current_config != grub_config {
+            println!("Removing {}", current_config);
+            grub_config_dir.remove_file(current_config.as_std_path())?;
```

Comment:
This is a clean up and not worth stopping the migration for. We can ignore the error and log it instead.

Reply:
https://docs.rs/openat-ext/latest/openat_ext/trait.OpenatDirExt.html#tymethod.remove_file_optional

Reply:
The if condition looks good